ipld/resolver/src/behaviour/membership.rs

```diff
@@ -70,14 +91,61 @@ pub struct Behaviour {
 }
 
 impl Behaviour {
+    pub fn new(nc: NetworkConfig, mc: Config) -> Result<Self, ConfigError> {
+        if nc.network_name.is_empty() {
+            return Err(ConfigError::InvalidNetwork(nc.network_name));
+        }
+        let membership_topic = Topic::new(format!("{}/{}", PUBSUB_MEMBERSHIP, nc.network_name));
+
+        let mut gossipsub_config = GossipsubConfigBuilder::default();
+        // Set the maximum message size to 2MB.
+        gossipsub_config.max_transmit_size(2 << 20);
+        gossipsub_config.message_id_fn(|msg: &GossipsubMessage| {
+            let s = blake2b_256(&msg.data);
+            MessageId::from(s)
+        });
+
+        let gossipsub_config = gossipsub_config
+            .build()
+            .map_err(|s| ConfigError::InvalidGossipsubConfig(s.into()))?;
+
+        let mut gossipsub = Gossipsub::new(
+            MessageAuthenticity::Signed(nc.local_key.clone()),
+            gossipsub_config,
+        )
+        .map_err(|s| ConfigError::InvalidGossipsubConfig(s.into()))?;
+
+        gossipsub
+            .with_peer_score(
+                scoring::build_peer_score_params(membership_topic.clone()),
+                scoring::build_peer_score_thresholds(),
```

Comment:
We may need to explore additional config parameters to avoid extreme load. In Go impl of Gossipsub there's the concept of `TopicValidators` that I don't know if in its Rust counterpart are needed, let's consider it for the future

Reply:
Definitely, the Forest stuff is [rather elaborate](https://github.com/ChainSafe/forest/blob/v0.6.0/node/forest_libp2p/src/gossip_params.rs), a copy of Lotus. 

I just had no idea how Filecoin specific it is. For example it has rates on the expected number of published messages, which is different for us, so it's obviously not something we can copy in, but the rest, I can't say.